x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -5565,7 +5565,353 @@ public void testToDatePeriodToTimeDurationWithField() {
         assertEquals("1:60: argument of [to_timeduration(x)] must be a constant, received [x]", e.getMessage().substring(header.length()));
     }
 
-    private Literal nullOf(DataType dataType) {
-        return new Literal(Source.EMPTY, null, dataType);
+    public void testCheckPercentileFoldableSecondArgument() {
```

Comment:
Agree. It must be a ValidatableTests kind of class, because these tests are specific to this approach of performing the validation. I've moved them to a new class. We need a followup to clean the rest of the "Verification" exception checks.